pkgs/tools/misc/pridefetch/default.nix

```diff
@@ -0,0 +1,51 @@
+{ lib, stdenv, fetchFromGitHub, python3, zip }: let
+  version = "1.0.0";
+  sha256 = "sha256-/o4er8bO/3HUFXzP+sC+5DYv9EwmxW05o1RT5fEulEg=";
+
+  pname = "pridefetch";
+  src = fetchFromGitHub {
+    owner = "SpyHoodle";
+    repo = pname;
+    rev = "v" + version;
+    inherit sha256;
+  };
+in stdenv.mkDerivation {
+  inherit pname version src;
+  nativeBuildInputs = [
+    zip
+  ];
+  buildInputs = [
+    (python3.withPackages (pythonPackages: with pythonPackages; [
+      distro
+    ]))
+  ];
+  buildPhase = ''
+    runHook preBuild
+    pushd src
+    zip -r ../pridefetch.zip ./*
+    popd
+    echo '#!/usr/bin/env python' | cat - pridefetch.zip > pridefetch
```

Comment:
Is this fixed by patchShebangs? Otherwise the package may not work at all if python is not installed. Maybe try with nix-shell --pure to see if it works well in that environment?

Reply:
Also, the dance with zipping still looks a bit weird to me. Why is it needed again?

Reply:
> Is this fixed by patchShebangs?

Yes

Reply:
> Also, the dance with zipping still looks a bit weird to me. Why is it needed again?

we have multiple python files in our project, and we can't just add a shebang to the top of our entry point file and be done with it as otherwise either the structure would change or we'd have to move all of the files into the bin directory. Zipping allows us to have a single file that contains all of the code and can be executed by python